### ISSUE TITLE
Stream error when saving an asset to an S3 bucket

### ISSUE THREAD

**Issue by davidrodriguezv** [pimcore/pimcore#6339]
# Bug Report

### Expected behavior
Assets are saved to the S3 bucket and the corresponding element is registered in the database without errors.

### Actual behavior
This message is showing on the admin interface everytime I'm trying to save an asset.
![MicrosoftTeams-image](https://user-images.githubusercontent.com/47228460/80996702-1f4f0e00-8df5-11ea-9ea4-d7515daa03bd.png)

After saving or creating an asset I found that the asset was uploaded/updated on the S3 bucket but the UI kept showing the previous error. 
When creating a new asset I found that a row was not created in the asset table. 
While debugging the logs showed that the database was being rolled back after the error was thrown. 

This is the stack trace of the error:

![MicrosoftTeams-image(2)](https://user-images.githubusercontent.com/47228460/80996718-237b2b80-8df5-11ea-833c-dd40462df1fc.png)
![MicrosoftTeams-image(1)](https://user-images.githubusercontent.com/47228460/80996725-25dd8580-8df5-11ea-8ded-cdd610ae0f0d.png)

### Steps to reproduce
Setup:
- Amazon S2 for Pimcore with Apache
- Amazon S3 for Assets as per the documentation
https://pimcore.com/docs/master/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/Amazon_AWS_Setup/Amazon_AWS_S3_Setup.html

### Code example
I found that the result of the stream_get_meta_data method was returning a closed resource.
I added the following validation to the vendor/pimcore/pimcore/models/Asset.php class on the getStream method.

get_resource_type($this->stream) != "stream"

**Original Code** ([source](https://github.com/pimcore/pimcore/blob/master/models/Asset.php))
```php
    /**
     * @return resource
     */
    public function getStream()
    {
        if ($this->stream) {
            $streamMeta = stream_get_meta_data($this->stream);
            if (!@rewind($this->stream) && $streamMeta['stream_type'] === 'STDIO') {
                $this->stream = null;
            }
        }

        if (!$this->stream && $this->getType() != 'folder') {
            if (file_exists($this->getFileSystemPath())) {
                $this->stream = fopen($this->getFileSystemPath(), 'r', false, File::getContext());
            } else {
                $this->stream = tmpfile();
            }
        }

        return $this->stream;
    }
```

**Fix**
```php
    /**
     * @return resource
     */
    public function getStream()
    {
        if ($this->stream) {
           if(get_resource_type($this->stream) != "stream"){
	    	    $this->stream = null;
            }else{
                $streamMeta = stream_get_meta_data($this->stream);
                if (!@rewind($this->stream) && $streamMeta['stream_type'] === 'STDIO') {
                    $this->stream = null;
                }
            }
        }

        if (!$this->stream && $this->getType() != 'folder') {
            if (file_exists($this->getFileSystemPath())) {
                $this->stream = fopen($this->getFileSystemPath(), 'r', false, File::getContext());
            } else {
                $this->stream = tmpfile();
            }
        }
	
        return $this->stream;
    }
```

**Comment by putzflorian** [comment]
works also for me 👍

**Comment by dvesh3** [comment]
fixed by #7345
